File: objc/Mediator/Mediator/ListBox.m
#import "ListBox.h"

@implementation ListBox

-(instancetype)initWithMediator:(id<DialogBoxMediator>)med {
    if(self = [super initWithMediator:med]) {
        _items = [[NSMutableArray alloc] init];
        _selectedIndex = -1;
    }
    return self;
}

-(void)addItem:(NSString*)item {
    [_items addObject:item];
}

-(void)selectItem:(int)index {
    if(index != _selectedIndex) {
        _selectedIndex = index;
        [self.med onChange:self];
    }
}

-(NSString*)getSelectedItem {
    return _items[_selectedIndex];
}

-(NSString*)toString {
    return [[NSString alloc] initWithFormat:@"Selected Item: %@", [_items objectAtIndex:_selectedIndex]];
}

@end

File: swift/Mediator/Mediator/ListBox.swift
import Foundation

public class ListBox : Widget {
    private var items: [String] = [String]()
    private var selectedIndex: Int = -1
    
    func addItem(_ item: String) -> Void {
        items.append(item)
    }
    
    func selectItem(_ index: Int) -> Void {
        if index != selectedIndex {
            selectedIndex = index
            self.mediator.onChange(self)
        }
    }
    
    func getSelectedItem() -> String {
        return items[selectedIndex]
    }
    
    func toString() -> String {
        return String("Selected Item: \(items[selectedIndex])")
    }
}
